File: samples/Tutorial/Movies.Test/DbTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.IO;
using System.Reflection;
using System.Threading.Tasks;

namespace Movies
{
    [TestClass]
    public class DbTests
    {
        private static string GetConnectionString()
        {
            string mdfFilename = "EmptyDb.mdf";
            string outputFolder = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
            string attachDbFilename = Path.Combine(outputFolder, mdfFilename);
            return string.Format(@"Data Source=(localdb)\MSSQLLocalDB;AttachDbFilename=""{0}"";Integrated Security=True", attachDbFilename);
        }

        private static Db CreateDb()
        {
            return new Db(GetConnectionString());
        }

        [TestMethod]
        public async Task Db_GetMovies()
        {
            using (var db = await MockMovie.CreateAsync(CreateDb()))
            {
                var result = await db.GetMoviesAsync("comedy");
                Assert.AreEqual(3, result.Count);
            }

            using (var db = await MockMovie.CreateAsync(CreateDb()))
            {
                var result = await db.GetMoviesAsync("ghost");
                Assert.AreEqual(2, result.Count);
            }
        }

        [TestMethod]
        public async Task Db_GetMovie()
        {
            using (var db = await MockMovie.CreateAsync(CreateDb()))
            {
                var result = await db.GetMovieAsync(1);
                Assert.AreEqual(1, result.Count);
            }
        }
    }
}

File: samples.vb/Tutorial/Movies.Test/DbTests.vb
Imports System.IO
Imports System.Reflection
Imports Microsoft.VisualStudio.TestTools.UnitTesting

<TestClass>
Public Class DbTests
    Private Shared Function GetConnectionString() As String
        Dim mdfFilename As String = "EmptyDb.mdf"
        Dim outputFolder As String = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)
        Dim attachDbFilename As String = Path.Combine(outputFolder, mdfFilename)
        Return String.Format("Data Source=(localdb)\MSSQLLocalDB;AttachDbFilename=""{0}"";Integrated Security=True", attachDbFilename)
    End Function

    Private Shared Function CreateDb() As Db
        Return New Db(GetConnectionString())
    End Function

    <TestMethod>
    Public Async Function Db_GetMovies() As Task
        Using db = Await MockMovie.CreateAsync(CreateDb())
            Dim result As Object = Await db.GetMoviesAsync("comedy")
            Assert.AreEqual(3, result.Count)
        End Using

        Using db = Await MockMovie.CreateAsync(CreateDb())
            Dim result As Object = Await db.GetMoviesAsync("ghost")
            Assert.AreEqual(2, result.Count)
        End Using
    End Function

    <TestMethod>
    Public Async Function Db_GetMovie() As Task
        Using db = Await MockMovie.CreateAsync(CreateDb())
            Dim result As Object = Await db.GetMovieAsync(1)
            Assert.AreEqual(1, result.Count)
        End Using
    End Function
End Class
